File: frontend/src/types/execution.ts
export interface Execution {
  id: string;
  job_id: string;
  job_name: string | null;
  started_at: string;
  finished_at: string | null;
  duration_ms: number | null;
  exit_code: number | null;
  status: 'running' | 'success' | 'failed' | 'timeout' | 'killed';
  stdout: string;
  stderr: string;
  triggered_by: string;
  retry_number: number;
  pid: number | null;
}

export interface ExecutionListResponse {
  items: Execution[];
  total: number;
  page: number;
  page_size: number;
}

File: backend/app/schemas/execution.py
from __future__ import annotations
from typing import Optional, List
from pydantic import BaseModel


class ExecutionResponse(BaseModel):
    id: str
    job_id: str
    job_name: Optional[str] = None
    started_at: str
    finished_at: Optional[str]
    duration_ms: Optional[int]
    exit_code: Optional[int]
    status: str
    stdout: str
    stderr: str
    triggered_by: str
    retry_number: int
    pid: Optional[int]

    model_config = {"from_attributes": True}


class ExecutionListResponse(BaseModel):
    items: List[ExecutionResponse]
    total: int
    page: int
    page_size: int
